core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -542,7 +496,54 @@ private[spark] class SecurityManager(
    * Gets the secret key.
    * @return the secret key as a String if authentication is enabled, otherwise returns null
    */
-  def getSecretKey(): String = secretKey
+  def getSecretKey(): String = {
+    if (isAuthenticationEnabled) {
+      Option(sparkConf.getenv(ENV_AUTH_SECRET))
```

Comment:
FWIW, in Mesos, we are planning on using the [Secrets primitives](https://github.com/apache/spark/blob/b3f9dbf48ec0938ff5c98833bb6b6855c620ef57/resource-managers/mesos/src/main/scala/org/apache/spark/deploy/mesos/config.scala#L26) to distribute `ENV_AUTH_SECRET`. This way Mesos and YARN can both use the same secret-generation code and only differ in the distribution of the secret. `SPARK_AUTH_SECRET_CONF` is already somewhat awkward because it has to be in the config.

Reply:
I can add a different, internal config for this is re-using `SPARK_AUTH_SECRET_CONF` is confusing. But I'm not too concerned about exposing this to the user code running the application; they can just as easily get that info from the UGI currently. Spark already redacts this kind of information when writing it to things like the event log, which would be one place where it might leak out.

Reply:
I changed things and now they work pretty much as before. It would be good to separate secret generation from distribution, but I'd rather do that separately.

Reply:
I agree SPARK_AUTH_SECRET_CONF is awkward and not really secure, when I initially did this , this is what was requested by other committers since standalone and mesos needed more security work around it anyway.

I don't follow how the MesosSecretConfig is going to be used fully.  Are these just regular spark configs passed around or distributed through mesos somehow?